libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/ManeuverProcessor.kt

```diff
@@ -390,19 +405,28 @@ internal object ManeuverProcessor {
                     list.forEach {
                         val directions = it.directions()
                         val active = it.active()
+                        val activeDirection: String? = profile?.let { value ->
+                            // In case of PROFILE_CYCLING [BannerComponent#activeDirection] is
+                            // unavailable and hence we need to fallback to [BannerText#modifier]
+                            if (active!! && value == DirectionsCriteria.PROFILE_CYCLING) {
+                                bannerInstruction.primary().modifier()
```

Comment:
As discussed, let's make this fallback dynamic depending on active direction availability and not hardcoding `PROFILE_CYCLING`.

Reply:
The property `activeDirection` is a `Nullable` property irrespective of profile. This behavior to hook onto the `activeDirection` if available for any profile otherwise fallback to `primary.modifier` has a limitation in a sense that I cannot define `fun isActiveDirectionPresent()` as the only way to check if it exists is to check if it is null or not. Since it is a `Nullable` property this check is incorrect. For ex: look at below responses

##### Driving profile
```
"sub": {
    "components": [
        {
            "active_direction": "slight left",
            "active": true,
            "directions": [
                "slight left"
            ],
            "type": "lane",
            "text": ""
        },
        {
            "active": false,
            "directions": [
                "right"
            ],
            "type": "lane",
            "text": ""
        }
    ]
```

##### Cycling profile
```
"sub": {
    "components": [
        {
            "active": true,
            "directions": [
                "slight left"
            ],
            "type": "lane",
            "text": ""
        },
        {
            "active": false,
            "directions": [
                "right"
            ],
            "type": "lane",
            "text": ""
        }
    ]
```
Do you have any other ideas?
/cc @LukasPaczos

Reply:
Why would this if-check not work?
```
if (active && active_direction == null) {
  bannerInstruction.primary().modifier()
}
```